### PR TITLE
[stable/kuberhealthy] kuberhealthy 2.0.0 release chart with helm 3

### PR DESCRIPTION
**Pull request by integrii**
This chart has been revamped to work with Helm 3 and our new upcoming release of Kuberhealthy 2.0.0.

Signed-off-by: Eric Greer <eric.greer@comcast.com>

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by integrii** [comment]
CircleCI says: `Error: apiVersion 'v2' is not valid. The value must be "v1"`

However, new charts created with Helm 3 using `helm create` come up with `apiVersion'v2'`.  I am guessing this is an issue with the build system here?

**Comment by integrii** [comment]
Maybe @davidkarlsen can let us know if Helm3 charts are supported yet?

**Comment by integrii** [comment]
I am going to change the apiVersion back to v1 just to get this PR merged for now.

**Review by ihoegen — APPROVED** [review]
/ok-to-test

**Comment by integrii** [comment]
circle ci may have failed because of a botched cleanup?

`I1115 20:02:47.651] Error: validation failed: unable to recognize "": no matches for kind "KuberhealthyCheck" in version "comcast.github.io/v1"`

**Review by jonnydawg — APPROVED** [review]
/lgtm

File: stable/kuberhealthy/templates/khcheck-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+kind: KuberhealthyCheck
+metadata:
+  name: deployment
+spec:
+  runInterval: 30m
+  timeout: 10m
+  podSpec:
+    containers:
+    - name: deployment
+      image: quay.io/comcast/deployment-check:1.0.0
```
Comment (jonnydawg) [inline stable/kuberhealthy/templates/khcheck-deployment.yaml]: this should be bumped to quay.io/comcast/deployment-check:1.0.1 !

**Review by jonnydawg — APPROVED** [review]

**Comment by integrii** [comment]
This CI run failed because the CRD was not recognized.  This was probably not recognized because the tests are not using Helm 3, which recognizes the new `crd` directory as a set of specs that need ran first.

File: stable/kuberhealthy/crds/khcheck.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  group: comcast.github.io
+  version: v1
+  scope: Namespaced
+  names:
+    plural: khchecks
+    singular: khcheck
+    kind: KuberhealthyCheck
+    shortNames:
+      - khc  
+
```
Comment (scottrigby) [inline stable/kuberhealthy/crds/khcheck.yaml:16]: Unnecessary extra line

File: stable/kuberhealthy/templates/clusterrole.yaml [hunk 3]
```diff
@@ -1,17 +1,62 @@
...
+    resources:
     - namespaces
     - componentstatuses
     - nodes
     verbs:
     - get
     - list
-    - watch
-  
+    - watch
```
Comment (scottrigby) [inline stable/kuberhealthy/templates/clusterrole.yaml]: Needs one newline at EOF

File: stable/kuberhealthy/templates/clusterrolebinding.yaml [hunk 4]
```diff
@@ -11,4 +11,4 @@ subjects:
 - kind: ServiceAccount
   name: {{ template "kuberhealthy.name" . }}
   namespace: {{ .Release.Namespace }}
-  
+
```
Comment (scottrigby) [inline stable/kuberhealthy/templates/clusterrolebinding.yaml]: Remove extra space on last line

File: stable/kuberhealthy/templates/khcheck-daemonset.yaml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      - patch
+      - update
+      - watch
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: daemonset-khcheck
+{{- end }}
+
```
Comment (scottrigby) [inline stable/kuberhealthy/templates/khcheck-daemonset.yaml:67]: Unnecessary extra line

File: stable/kuberhealthy/templates/khcheck-dns.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+            value: "kubernetes.default"
+        image: quay.io/comcast/dns-status-check:1.0.0
+        imagePullPolicy: IfNotPresent
+        name: main
+        resources:
+          requests:
+            cpu: 10m
+            memory: 50Mi
+{{- end }}
+
```
Comment (scottrigby) [inline stable/kuberhealthy/templates/khcheck-dns.yaml]: Unnecessary extra line

**Review by scottrigby — COMMENTED** [review]
Re helm 3 charts, see this recent README update: https://github.com/helm/charts#status-of-the-project

Also some newline and extra line fixes needed.

**Comment by scottrigby** [comment]
Also see https://github.com/helm/charts/issues/19008

**Comment by davidkarlsen** [comment]
Should this chart be deprecated in favour of https://github.com/Comcast/kuberhealthy/tree/master/deploy/helm/kuberhealthy ?
Maybe add the comcast repo to helm.hub: https://github.com/helm/hub/blob/master/Repositories.md ?

**Comment by integrii** [comment]
@davidkarlsen Good catch!  That chart in the main Kuberhealthy project is basically this one.  We run the script there in the same directory with it and contribute that as a chart here.

Now that helm 3 exists, I think having our own repository is much better - the release process here is hard!  I will make an issue on Kuberhealthy for this now.

**Pull request closed without merging**